diff --git a/install.js b/install.js
index 540e1e68a561fbd0b3af0438b0465e984a7d5b52..746990829b603092d182e011e576df854c9ef1fe 100755
--- a/install.js
+++ b/install.js
@@ -42,7 +42,10 @@ if (
 downloadArtifact({
   version,
   artifactName: 'electron',
-  force: process.env.force_no_cache === 'true',
+  // Joplin change: Since https://github.com/electron/get/pull/267, @electron/get
+  // seems to ignore the force option:
+  // force: process.env.force_no_cache === 'true',
+  cacheMode: process.env.ELECTRON_CACHE_MODE,
   cacheRoot: process.env.electron_config_cache,
   checksums:
     process.env.electron_use_remote_checksums || process.env.npm_config_electron_use_remote_checksums
