diff --git a/install.js b/install.js
index 540e1e68a561fbd0b3af0438b0465e984a7d5b52..a8982a02ae64c0a16960a894bc326ddab0a9e400 100755
--- a/install.js
+++ b/install.js
@@ -14,12 +14,15 @@ const { version } = require('./package');
 const platformPath = getPlatformPath();
 
 if (isInstalled()) {
+  console.log('Already installed!');
   process.exit(0);
 }
 
 const platform = process.env.ELECTRON_INSTALL_PLATFORM || process.env.npm_config_platform || process.platform;
 let arch = process.env.ELECTRON_INSTALL_ARCH || process.env.npm_config_arch || process.arch;
 
+console.log('Not installed yet', platform, arch);
+
 if (
   platform === 'darwin' &&
   process.platform === 'darwin' &&
@@ -77,9 +80,12 @@ function isInstalled() {
 
 // unzips and makes path.txt point at the correct executable
 function extractFile(zipPath) {
+  console.log('Artifact downloaded to', zipPath);
   const distPath = process.env.ELECTRON_OVERRIDE_DIST_PATH || path.join(__dirname, 'dist');
+  console.log('Dist path', distPath);
 
   return extract(zipPath, { dir: path.join(__dirname, 'dist') }).then(() => {
+    console.log('Extraction completed');
     // If the zip contains an "electron.d.ts" file,
     // move that up
     const srcTypeDefPath = path.join(distPath, 'electron.d.ts');
