diff --git a/lib/dispatcher/client-h1.js b/lib/dispatcher/client-h1.js
index c775cb988f4786fb906e246fcb2b654d0b2531e6..d97f667b2d783c06e6eb39f268c34070e096e566 100644
--- a/lib/dispatcher/client-h1.js
+++ b/lib/dispatcher/client-h1.js
@@ -6,6 +6,9 @@ const assert = require('node:assert')
 const util = require('../core/util.js')
 const { channels } = require('../core/diagnostics.js')
 const timers = require('../util/timers.js')
+// PATCH: Electron's setTimeout does not have an ".unref" method.
+const { setTimeout, clearTimeout } = require('node:timers');
+
 const {
   RequestContentLengthMismatchError,
   ResponseContentLengthMismatchError,
diff --git a/lib/util/timers.js b/lib/util/timers.js
index 14984d42ef2901af503b595002bc973bf069f9a4..045f3791a70486e445835ec388f2678cfb0f8572 100644
--- a/lib/util/timers.js
+++ b/lib/util/timers.js
@@ -1,5 +1,8 @@
 'use strict'
 
+// PATCH: Electron's setTimeout does not have an ".unref" method.
+const { setTimeout, clearTimeout } = require('node:timers');
+
 /**
  * This module offers an optimized timer implementation designed for scenarios
  * where high precision is not critical.
diff --git a/lib/web/fetch/index.js b/lib/web/fetch/index.js
index a0dd75df7a5224657507371439859a0d6cefdc66..173bdef9a4bfce4e464200b5c6421b87726d0a02 100644
--- a/lib/web/fetch/index.js
+++ b/lib/web/fetch/index.js
@@ -13,6 +13,7 @@ const {
 const { HeadersList } = require('./headers')
 const { Request, cloneRequest, getRequestDispatcher, getRequestState } = require('./request')
 const zlib = require('node:zlib')
+const { performance } = require('node:perf_hooks');
 const {
   makePolicyContainer,
   clonePolicyContainer,
